internal/controller/variantautoscaling_controller.go

```diff
@@ -400,19 +400,38 @@ func (r *VariantAutoscalingReconciler) SetupWithManager(mgr ctrl.Manager) error
 	// Initialize metrics
 	initMetricsEmitter()
 
-	// Configure Prometheus client using flexible configuration
-	// TODO: If configuration changes become a concern, implement a configuration watcher rather than per-cycle initialization.
-	prom_addr, err := r.getPrometheusConfig(context.Background())
+	// Configure Prometheus client using flexible configuration with TLS support
+	promConfig, err := r.getPrometheusConfig(context.Background())
 	if err != nil {
 		logger.Log.Warn("Failed to get Prometheus config, using default - ", "error: ", err)
 		// Use default as fallback
-		prom_addr = "http://prometheus-operated.inferno-autoscaler-monitoring.svc.cluster.local:9090"
+		promConfig = &interfaces.PrometheusConfig{
+			BaseURL: "http://kube-prometheus-stack-prometheus.inferno-autoscaler-monitoring.svc.cluster.local:9090",
+			Timeout: 30 * time.Second,
+		}
 	}
 
-	logger.Log.Info("Initializing Prometheus client -> ", "address: ", prom_addr)
-	promClient, err := api.NewClient(api.Config{
-		Address: prom_addr,
-	})
+	// Validate TLS configuration if enabled
+	if promConfig.TLS != nil && promConfig.TLS.EnableTLS {
+		if err := utils.ValidateTLSConfig(promConfig.TLS); err != nil {
+			logger.Log.Error(err, "TLS configuration validation failed, falling back to HTTP")
+			promConfig.TLS = nil
+			// Ensure BaseURL is HTTP if TLS fails
+			if len(promConfig.BaseURL) > 8 && promConfig.BaseURL[:8] == "https://" {
```

Comment:
can we remove this?

Reply:
Stripping base URL with slice indexes looks fragile to me.

Reply:
good catch, taken care of with a helper function